src/FadingCounter.cc

```diff
@@ -14,63 +14,53 @@
 
 #include <cmath>
 
-FadingCounter::FadingCounter(): horizon(-1), precision(10), delta(-1),
-    lastTime(0), total(0)
+void
+FadingCounter::clear()
 {
-    counters.reserve(precision);
-    while (counters.size() < static_cast<unsigned int>(precision))
-        counters.push_back(0);
-}
-
-void FadingCounter::clear()
-{
-    for (int i = 0; i < precision; ++i)
-        counters[i] = 0;
+    std::fill(counters.begin(), counters.end(), 0);
```

Comment:
Please simplify this code, similar to code this PR already uses to initialize `counters` during object construction:

```suggestion
    counters = {};
```

Reply:
Done in [c60bf4c](https://github.com/squid-cache/squid/pull/1782/commits/c60bf4cce828a7f5ac37876def190de9db4fa9e1)